### DISCUSSION TITLE
How to update dynamic column data from user input

### DISCUSSION THREAD

**Discussion by Tmcerlean** [prisma/prisma#14869] [Q&A]
Hi there,

I'm very new to using Prisma and was hoping somebody here would be able to assist with the below.

I am working on a web application which enables users to interact with a database (containing a long list of thousands of products and related information) via authenticated REST endpoints. 

![image](https://user-images.githubusercontent.com/31129724/185252089-371d8024-8146-4367-bde7-89461c51bd1c.png)

One of such endpoints enables the filtering of products through criteria defined on the front-end and submitted. I was planning on achieving this by using two steps:

**1. Filter Criteria**
- Column (e.g. title)
- Operator (contains / does not contain / is / is not blank / starts with / ends with)
- Column (e.g. Size) OR Text (e.g. small)

It also enables the filtered set of product information to be amended by a further set of (Action) criteria:

**2. Action Criteria**
- Column (e.g. title)
- Operator (e.g. set / prepend / append / delete from / replace)
- Column (e.g. Size) OR Text (e.g. small)

------

**My problem:**

I have been attempting to write a Prisma query that enables users to submit a column they want to edit and the associated new value for the field. From what I can understand from the documentation, it is not possible to do this due to a risk of SQL injection.

Consequently, I would need to write a function for all 70+ columns in my table and have multiple variations within it for each of the different potential operators which could be provided by the user. Below is an example of the Title field being filtered for products which do not contain the associated Size column (from the front-end table).

```ts
const products = await prisma.product.findMany({
    // Find products based upon submitted criteria
    where: {
        NOT: [
            {
                title: {
                    contains: req.body.size
                } 
            },
            {
                title: ""
            }
        ]
    },
})
```

I looked into writing some raw Postgres queries for what I needed to do, but I continually ran into warnings that any potential solution (using dynamic columns) would lead to security vulnerabilities due to SQL injection. Here was one of such queries related to the second (Action) filtering step mentioned above:
```ts
const result = await prisma.$executeRaw`UPDATE "Product" SET ${req.body.column} = ${req.body.value} WHERE id = ${req.body.id};
```
Which would need to be re-written as:
```ts
const result = await prisma.$executeRaw`UPDATE "Product" SET title = ${req.body.value} WHERE id = ${req.body.id};
```
Is what I am doing achievable in a less verbose way, or is it the way I have identified the only way which is secure?

Thank you - and apologies for the lengthy query!

**Comment by austincrim** [comment]
Hey @Tmcerlean 👋,

I'm not sure I grasp all the constraints of your question, but regarding your example Postgres queries above, have you tried something like this:
```ts
const result = await prisma.product.update({
  where: { id: req.body.id },
  data: { [req.body.column]: req.body.value }
})
```

**Reply by vladi-stevanovic** [reply]
👋 Hello @Tmcerlean I'm going through pending GitHub Discussions to check if you need any help. Is this issue still open?
Please let us know if we can investigate it further or if Austin's solution helped, feel free to "Mark as answer" if this discussion is now solved!

It would be super useful for other users facing this problem! 🙌

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
